bitsail-connectors/connector-selectdb/src/main/java/com/bytedance/bitsail/connector/selectdb/committer/SelectdbCommitter.java

```diff
@@ -0,0 +1,178 @@
+import com.bytedance.bitsail.connector.selectdb.error.SelectdbErrorCode;
+import com.bytedance.bitsail.connector.selectdb.http.HttpPostBuilder;
+import com.bytedance.bitsail.connector.selectdb.http.HttpUtil;
+import com.bytedance.bitsail.connector.selectdb.http.ResponseUtil;
+import com.bytedance.bitsail.connector.selectdb.http.model.BaseResponse;
+import com.bytedance.bitsail.connector.selectdb.http.model.CopyIntoResp;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.NoArgsConstructor;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.bytedance.bitsail.connector.selectdb.sink.uploadload.LoadStatus.FAIL;
+import static com.bytedance.bitsail.connector.selectdb.sink.uploadload.LoadStatus.SUCCESS;
+import static org.apache.http.HttpStatus.SC_OK;
+
+@NoArgsConstructor
+public class SelectdbCommitter implements WriterCommitter<SelectdbCommittable> {
+  private static final Logger LOG = LoggerFactory.getLogger(SelectdbCommitter.class);
+  private static final String COMMIT_PATTERN = "http://%s/copy/query";
+  private final ObjectMapper objectMapper = new ObjectMapper();
+  private SelectdbOptions selectdbOptions;
+  private int maxRetry;
+  private transient CloseableHttpClient httpClient;
+
+  public SelectdbCommitter(SelectdbOptions selectdbOptions, SelectdbExecutionOptions executionOptions) {
+    this.selectdbOptions = selectdbOptions;
+    this.maxRetry = executionOptions.getMaxRetries();
+  }
+
+  @Override
+  public List<SelectdbCommittable> commit(List<SelectdbCommittable> committables) throws IOException {
+    for (SelectdbCommittable committable : committables) {
+      try {
+        // TODO add a open method init it
+        this.httpClient = new HttpUtil().getHttpClient();
+
+        commitTransaction(committable);
+      } finally {
+        close();
+      }
+    }
+    return Collections.emptyList();
+  }
+
+  private void commitTransaction(SelectdbCommittable committable) throws IOException {
+    long start = System.currentTimeMillis();
+    String hostPort = committable.getHostPort();
+    String clusterName = committable.getClusterName();
+    String copySQL = committable.getCopySQL();
+    LOG.info("commit to cluster {} with copy sql: {}", clusterName, copySQL);
+
+    int retry = 0;
+    CloseableHttpResponse response = null;
+    String loadResult = "";
+    boolean success = false;
+    String reasonPhrase = null;
+    int statusCode = -1;
+    while (retry++ <= maxRetry) {
```

Comment:
Have you ever considered to use `com.github.rholder.retry.Retryer` instead of "while" loop?
It's a convenient tool.

Reply:
I tried to use `com.github.rholder.retry.Retryer`, and found that it cannot be serialized, and I need to register Retryer every time when I use it.

 I think it is better not to change it.